### PR TITLE
Update minimum supported version to 2018.4

### PR DESCRIPTION
**Pull request by vkovec**

### CODE REVIEW

**Review by benoithudson — APPROVED** [review]

File: build.py [hunk 1]
```diff
@@ -33,7 +32,8 @@
     shell = True
     cmake_exe = 'cmake.exe'
 else:
-    if platform.linux_distribution()[0].startswith('CentOS'):
+    import distro
+    if distro.linux_distribution(full_distribution_name=False)[0].startswith('centos'):
         # cmake is CMake2 on centos, need to be explicit.
         cmake_exe = 'cmake3'
```
Comment (benoithudson) [inline build.py:38]: Can you remove the linux distro test and parse `cmake --version`? Or check if cmake3 exists and if not default to just cmake?

**Review by benoithudson — CHANGES_REQUESTED** [review]

**Comment by benoithudson** [comment]
And to use pip on yamato you need to set an environment variable so it can find the repository, see the python package. It got disconnected from the global pypi repository because of a security issue.

File: build.py [hunk 2]
```diff
@@ -33,7 +32,9 @@
     shell = True
     cmake_exe = 'cmake.exe'
 else:
-    if platform.linux_distribution()[0].startswith('CentOS'):
+    # check if we should use cmake or cmake3
+    cmake_version = subprocess.check_output(["cmake", "--version"], shell=False)
+    if cmake_version.startswith(b'cmake3'):
```
Comment (benoithudson) [inline build.py:37]: on my mac it's `cmake version 3.14.0`

Comment (benoithudson) [reply build.py:37]: but maybe I misunderstood this line's logic

Comment (vkovec) [reply build.py:37]: yes this should work no?

I'm checking if the result is `cmake3 version ...` vs `cmake ...`

Comment (vkovec) [reply build.py:37]: or maybe I should just parse the first part instead of checking if it starts with it

Comment (benoithudson) [reply build.py:38]: I don't understand this code.

If cmake is returning "cmake3 version 3.x.y" then why do you need to switch to cmake3 -- you're already using it.

If it's returning "cmake version 2.x.y" then you accept it even though it won't work.

I suspect the problem this code is trying to deal with just doesn't exist anymore so you could simply use cmake and go on with life.

**Pull request merged**